### DISCUSSION TITLE
Relation does not show on edit page

### DISCUSSION THREAD

**Discussion by rubabnaziri** [filamentphp/filament#2122] [General]
I have a simple application with Users, Roles, Permissions, Groups (Teams in filament)
Users have Roles, 
Roles have Permissions, 
Groups have Roles and Users. 

On my **User** Edit page I can see Group and Role table.
However On my **Roles** Edit page I do not see the Permissions table, i.e. I can't see what permissions are attached to a role. 

Can someone help with this?

**RoleResource**
`<?php

namespace App\Filament\Resources;

use App\Filament\Resources\RoleResource\Pages;
use App\Filament\Resources\RoleResource\RelationManagers;
use Filament\Forms\Components\TextInput;
use Filament\Resources\Form;
use Filament\Resources\Resource;
use Filament\Resources\Table;
use Filament\Tables\Columns\TextColumn;
use Spatie\Permission\Models\Role as ModelsRole;

class RoleResource extends Resource
{
    protected static ?string $model = ModelsRole::class;

    protected static ?string $navigationIcon = 'heroicon-o-collection';

    protected static ?string $navigationGroup = 'Settings';

    public static function form(Form $form): Form
    {
        return $form
            ->schema([
                TextInput::make('name')
                    ->required()
                    ->maxLength(255),
                TextInput::make('guard_name')
                    ->required()
                    ->maxLength(255),
            ]);
    }

    public static function table(Table $table): Table
    {
        return $table
            ->columns([
                TextColumn::make('id'),
                TextColumn::make('name'),
                TextColumn::make('guard_name'),
            ])
            ->filters([
                //
            ]);
    }

    public static function getRelations(): array
    {
        return [
            RelationManagers\PermissionsRelationManager::class,
        ];
    }

    public static function getPages(): array
    {
        return [
            'index' => Pages\ListRoles::route('/'),
            'create' => Pages\CreateRole::route('/create'),
            'edit' => Pages\EditRole::route('/{record}/edit'),
        ];
    }
}
`

**PermissionsRelationManager**
`<?php

namespace App\Filament\Resources\RoleResource\RelationManagers;

use Filament\Forms;
use Filament\Resources\Form;
use Filament\Resources\RelationManagers\BelongsToManyRelationManager;
use Filament\Resources\Table;
use Filament\Tables;
use Illuminate\Database\Eloquent\Model;

class PermissionsRelationManager extends BelongsToManyRelationManager
{
    protected static string $relationship = 'permissions';

    protected static ?string $recordTitleAttribute = 'name';

    //There shouldn't be an option to create or delete permissions from roles page 
    protected function canDeleteAny(): bool { return false; }
    protected function canDelete(Model $record): bool { return false; }  
    protected function canCreate(): bool { return false; }

    public static function form(Form $form): Form
    {
        return $form
            ->schema([
                Forms\Components\TextInput::make('name')->required(),
            ]);
    }

    public static function table(Table $table): Table
    {
        return $table
            ->columns([
                Tables\Columns\TextColumn::make('id'),
                Tables\Columns\TextColumn::make('name'),
            ])
            ->filters([
            ]);
    }

    public static function attachForm(Form $form): Form
    {
        return $form
            ->schema([
                static::getAttachFormRecordSelect(),
            // ...
            ]);
    }
}
`

**Comment by danharrin** [comment]
If you have a PermissionPolicy, make sure that the user can `viewAny()` - otherwise the relation manager will not show on the edit page

**Reply by Quadrubo** [reply]
Why is the PermissionPolicy responsible for that? Shouldn't the PermissionPolicy provide the authorization for the Permission Resource?

**Reply by Z3d0X** [reply]
@Quadrubo  Yes `PermissionPolicy` will provide authorization for `PermissionResource` (that uses `Permission` Model). 

`PermissionPolicy` will also be used to authorize any RelationManagers that uses `Permission` Model.

**Reply by Quadrubo** [reply]
My relationmanager doesn't use the Permission Model but still needs view_any_permission to work.
